tests/test_Summary.cpp

```diff
@@ -0,0 +1,365 @@
+
+    BOOST_CHECK_CLOSE( 30.0 + 30.1, ecl_sum_get_well_var( resp, 0, "W_3", "WLPR" ), 1e-5 );
+    BOOST_CHECK_CLOSE( (30.0 + 30.1) / day, ecl_sum_get_well_var( resp, 0, "W_3", "WLPT" ), 1e-5 );
+    BOOST_CHECK_CLOSE( (30.0 + 30.1), ecl_sum_get_well_var( resp, 1, "W_3", "WLPR" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * (30.0 + 30.1) / day, ecl_sum_get_well_var( resp, 1, "W_3", "WLPT" ), 1e-5 );
+
+    rmfiles( fname );
+}
+
+BOOST_AUTO_TEST_CASE(W_WOG_PRH) {
+    const auto deck = Parser().parseFile( path, ParseContext() );
+    EclipseState es( deck, ParseContext() );
+    Summary summary( *deck, es );
+    const auto wells = result_wells();
+
+    std::string fname = "sum_test_W_WOG_PRH";
+
+    out::Summary writer( es, summary, fname );
+    writer.add_timestep( 1, 1, es, wells );
+    writer.write();
+
+    auto res = readsum( fname );
+    const auto* resp = res.get();
+
+    BOOST_CHECK_CLOSE( 10, ecl_sum_get_well_var( resp, 0, "W_1", "WWPRH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 20, ecl_sum_get_well_var( resp, 0, "W_2", "WWPRH" ), 1e-5 );
+
+    BOOST_CHECK_CLOSE( 10.1, ecl_sum_get_well_var( resp, 0, "W_1", "WOPRH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 20.1, ecl_sum_get_well_var( resp, 0, "W_2", "WOPRH" ), 1e-5 );
+
+    BOOST_CHECK_CLOSE( 10.2, ecl_sum_get_well_var( resp, 0, "W_1", "WGPRH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 20.2, ecl_sum_get_well_var( resp, 0, "W_2", "WGPRH" ), 1e-5 );
+
+    rmfiles( fname );
+}
+
+BOOST_AUTO_TEST_CASE(W_WOG_PTH) {
+    const auto deck = Parser().parseFile( path, ParseContext() );
+    EclipseState es( deck, ParseContext() );
+    Summary summary( *deck, es );
+    const auto wells = result_wells();
+
+    std::string fname = "sum_test_W_WOG_PRH";
+
+    out::Summary writer( es, summary, fname );
+    writer.add_timestep( 1, 1, es, wells );
+    writer.add_timestep( 2, 1, es, wells );
+    writer.write();
+
+    auto res = readsum( fname );
+    const auto* resp = res.get();
+
+    BOOST_CHECK_CLOSE( 2 * 10.0 / day, ecl_sum_get_well_var( resp, 1, "W_1", "WWPTH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * 20.0 / day, ecl_sum_get_well_var( resp, 1, "W_2", "WWPTH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * 10.1 / day, ecl_sum_get_well_var( resp, 1, "W_1", "WOPTH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * 20.1 / day, ecl_sum_get_well_var( resp, 1, "W_2", "WOPTH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * 10.2 / day, ecl_sum_get_well_var( resp, 1, "W_1", "WGPTH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * 20.2 / day, ecl_sum_get_well_var( resp, 1, "W_2", "WGPTH" ), 1e-5 );
+
+    rmfiles( fname );
+}
+
+BOOST_AUTO_TEST_CASE(WRITE) {
+    const auto deck = Parser().parseFile( path, ParseContext() );
+    EclipseState es( deck, ParseContext() );
+    Summary summary( *deck, es );
+    const auto wells = result_wells();
+
+    out::Summary writer( es, summary );
+    writer.add_timestep( 1, 1, es, wells );
+    writer.add_timestep( 2, 1, es, wells );
+    writer.write();
+}
```

Comment:
The testing is *very good and extensive* on the various variables, actual values and so on - but maybe a bit lacking on the *time direction*? The following `ecl_sum` functions can be used to test/verify some aspects of the time direction:

```
/* Complete summary case: */
time_t ecl_sum_get_start_time( const ecl_sum_type * );
time_t ecl_sum_get_data_start( const ecl_sum_type * );
time_t ecl_sum_get_end_time(const ecl_sum_type *);
double ecl_sum_get_sim_length(const ecl_sum_type*); // days


/* Single timestep */
time_t ecl_sum_iget_sim_time( const ecl_sum_type * , int );
double ecl_sum_iget_sim_days( const ecl_sum_type *, int );

/* Report */
time_t ecl_sum_get_report_time( const ecl_sum_type * , int report_step );
```